etc/inc/util.inc

```diff
@@ -1420,58 +1420,43 @@ function exec_command($command) {
 	return(implode("\n", $output));
 }
 
-/* wrapper for exec() */
-function mwexec($command, $mute = false, $clearsigmask = false) {
+/* wrapper for exec() 
+   Executes in background or foreground.
+   For background execution, returns PID of background process to allow calling code control */
+function mwexec($command, $nologentry = false, $clearsigmask = false, $background = false) {
 	global $g;
-
+	$outputarray = array();
+	$retval = 0;
+	
 	if ($g['debug']) {
 		if (!$_SERVER['REMOTE_ADDR']) {
-			echo "mwexec(): $command\n";
+			echo "mwexec(): $command" . ($background ? " [BG]":"") . "\n";
 		}
 	}
-	$oarr = array();
-	$retval = 0;
-
 	if ($clearsigmask) {
 		$oldset = array();
 		pcntl_sigprocmask(SIG_SETMASK, array(), $oldset);
 	}
-	$garbage = exec("$command 2>&1", $oarr, $retval);
+
+	if ($background)
```

Comment:
Please add curly brackets, @phil-davis did a huge job recently adding it and fixing indent